apps/token-manager/app/src/components/Panels/TokenPanelContent.js

```diff
@@ -17,233 +18,279 @@ import LocalIdentitiesAutoComplete from '../LocalIdentitiesAutoComplete/LocalIde
-      amountField: {
+      setHolderField({ ...holderField, value, error: null })
+      setAmountField({
         ...amountField,
         max: formatBalance(maxAmount, tokenDecimalsBase, tokenDecimals),
         warning:
           maxAmount.isZero() &&
           (mode === 'assign'
             ? `
-              The maximum amount of tokens that can be assigned has already been
-              reached
+              The maximum amount of tokens that can be assigned has already
+              been reached.
             `
             : `
-              This account doesn’t have any tokens to remove
+              This account doesn’t have any tokens to remove.
             `),
-      },
-    }))
-  }
-  handleAmountChange = event => {
-    const { amountField } = this.state
-    this.setState({
-      amountField: { ...amountField, value: event.target.value },
-    })
-  }
-  handleHolderChange = value => {
-    this.updateHolderAddress(this.props.mode, value)
-  }
-  handleSubmit = event => {
-    event.preventDefault()
-    const { mode } = this.props
-    const holderAddress = this.filteredHolderAddress()
+      })
+    },
+    [
+      amountField,
+      getHolderBalance,
+      holderField,
+      maxAccountTokens,
+      mode,
+      tokenDecimals,
+      tokenDecimalsBase,
+    ]
+  )
+
+  const updateAmount = useCallback(
+    value => {
+      setAmountField({ ...amountField, value })
+    },
+    [amountField]
+  )
 
-    const holderError = !isAddress(holderAddress)
-      ? `
-        ${mode === 'assign' ? 'Recipient' : 'Account'}
-        must be a valid Ethereum address
-      `
-      : null
+  const validateFields = useCallback(() => {
+    const holderAddress = holderField.value.trim()
+    const holderError = isAddress(holderAddress)
+      ? null
+      : mode === 'assign'
+      ? 'The recipient must be a valid Ethereum address.'
+      : 'The account must be a valid Ethereum address.'
 
-    // Error
     if (holderError) {
-      this.setState(({ holderField }) => ({
-        holderField: { ...holderField, error: holderError },
-      }))
-      return
+      setHolderField({ ...holderField, error: holderError })
+      return null
     }
 
-    // Update tokens
-    this.props.onUpdateTokens({
-      mode,
-      amount: this.filteredAmount(),
-      holder: holderAddress,
-    })
+    return { holder: filterHolder(), amount: filterAmount() }
+  }, [mode, holderField, filterHolder, filterAmount])
+
+  return {
+    amountField,
+    errorMessage,
+    holderField,
+    submitDisabled,
+    updateAmount,
+    updateHolder,
+    validateFields,
+    warningMessage,
   }
-  render() {
-    const { holderField, amountField } = this.state
-    const { mode, tokenDecimals } = this.props
+}
+
+function TokenPanelContent({
+  getHolderBalance,
+  holderAddress,
+  maxAccountTokens,
+  mode,
+  onUpdateTokens,
+  tokenDecimals,
+  tokenDecimalsBase,
+}) {
+  const holderInputRef = useSidePanelFocusOnReady()
```

Comment:
I think it should be fairly easy to do with the `useSidePanel()` hook, but it would be nice if we instead focused on the amount input if the `holderAddress` was already populated.

Reply:
Done in https://github.com/aragon/aragon-apps/pull/1046/commits/5535c0b17d9e9b4970a08680f60679366ed91383